src/benchmarks/micro/ValuesGenerator.cs

```diff
@@ -87,12 +94,40 @@ private static T GenerateValue<T>(Random random)
                 return (T)(object)random.Next();
             if (typeof(T) == typeof(double))
                 return (T)(object)random.NextDouble();
+            if (typeof(T) == typeof(float))
+                return (T)(object)(float)random.NextDouble();
             if (typeof(T) == typeof(string))
                 return (T) (object) GenerateRandomString(random, 1, 50);
             
             throw new NotImplementedException($"{typeof(T).Name} is not implemented");
         }
 
+        private static T GenerateValue<T>(Random random, T minValue, T maxValue)
+        {
+            // This is currently used for many of the System.Math and System.MathF perf tests.
+            //
+            // These functions often have special handling in the JIT to either cache the computed
+            // result or to perform constant folding. This allows us to generate a unique input
+            // per iteration that is within the allowed input domain for a given method.
+
+            if (typeof(T) == typeof(int))
+                return (T)(object)random.Next((int)(object)minValue, (int)(object)maxValue);
+            if (typeof(T) == typeof(double))
+                return (T)(object)BitConverter.Int64BitsToDouble(
+                    (long)random.Next(
+                        (int)(BitConverter.DoubleToInt64Bits((double)(object)minValue) >> 32),
+                        (int)(BitConverter.DoubleToInt64Bits((double)(object)maxValue) >> 32)
+                    ) << 32
+                );
+            if (typeof(T) == typeof(float))
+            {
+                int result = random.Next(Unsafe.As<T, int>(ref minValue), Unsafe.As<T, int>(ref maxValue));
```

Comment:
maybe we could take the double result and cast it to float?  `(T)(object)(float)GenerateValue<double>()`

Reply:
Same comment as above.